data_scrapers/san_francisco.py

```diff
@@ -0,0 +1,268 @@
+        params = {'case_disposition': 'Death',
+                '$select': 'date,sum(case_count) as deaths', '$group': 'date', '$order': 'date'}
+        series = self.request(resource_id, params=params)
+        death_series = []
+        # convert date from ISO string to 'yyyy-mm-dd'. convert number strings to int.
+        # calculate daily cumulative
+        cumul = 0
+        for entry in series:
+            entry["date"] = entry["date"][0:10]
+            entry["deaths"] = int(entry["deaths"])
+            cumul += entry["deaths"]
+            entry["cumul_deaths"] = cumul
+            death_series.append(entry)
+        return death_series
+
+    # Daily count of tests with count of positive tests
+    # Note that SF county does not include pending tests, and does not directly report negative tests or cumulative tests.
+    def get_tests_series(self) -> List[Dict]:
+        """Get tests by day, order by date ascending"""
+        resource_id = self.resource_ids['tests']
+        test_series = []
+        params = {'$order': 'result_date'}
+        series = self.request(resource_id, params=params)
+
+        # parse source series into out series, calculating cumulative values
+        # Counter is from the built-in `collections` module.
+        totals:Counter = Counter()
+        for entry in series:
+            out_entry = dict(date=entry["result_date"][0:10],
+                            tests=int(entry["tests"]),
+                            positive=int(entry["pos"]))
+            out_entry['negative'] = out_entry["tests"] - out_entry["positive"]
+            totals.update(out_entry)
+            out_entry.update(cumul_tests=totals['tests'],
+                            cumul_pos=totals['positive'],
+                            cumul_neg=totals['negative'])
+            test_series.append(out_entry)
+        return test_series
+
+    def get_age_table(self) -> List[Dict]:
+        """Get cases by age"""
+        resource_id = self.resource_ids['age_gender']
+        # Dict of target_label:source_label for lookups
+        AGE_KEYS = {"18_and_under": "under 18", "81_and_older": "81+", "18_to_30": "18-30", "31_to_40": "31-40", "41_to_50": "41-50",  "51_to_60": "51-60", "61_to_70":"61-70", "71_to_80": "71-80"}
+
+
+        params = {'$select': 'age_group, sum(confirmed_cases)', '$order': 'age_group', '$group': 'age_group'}
+        data = self.request(resource_id, params=params)
+
+        # structure age_table per our README
+        age_table : List[Dict[str,Any]] = [
+            {"group": "18_and_under", "raw_count": -1},
+            {"group": "18_to_30", "raw_count": -1},
+            {"group": "31_to_40", "raw_count": -1},
+            {"group": "41_to_50", "raw_count": -1},
+            {"group": "51_to_60", "raw_count": -1},
+            {"group": "61_to_70", "raw_count": -1},
+            {"group": "71_to_80", "raw_count": -1},
+            {"group": "81_and_older", "raw_count": -1}
+	    ]
```

Comment:
Kind of like I noted for Alameda county, you don’t need to write all this out if you make sure `AGE_KEYS` is in order.